velox/experimental/cudf/exec/ExpressionEvaluator.cpp

```diff
@@ -649,6 +709,29 @@ void addPrecomputedColumns(
           stream,
           cudf::get_current_device_resource_ref());
       input_table_columns.emplace_back(std::move(newColumn));
+    } else if (ins_name == "cardinality") {
+      auto newColumn = cudf::lists::count_elements(
```

Comment:
Just checking this behavior since I'm not certain what result is desired here. `cudf::lists::count_elements` returns a _size_, not a cardinality. It's the total number of elements, not the number of unique elements.

Reply:
Presto's documentation says "Returns the cardinality (size) of the array x." so we're OK.